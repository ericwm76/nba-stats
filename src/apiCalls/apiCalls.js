const baseUrl = 'https://www.balldontlie.io/api/v1'

export const searchByName = async (firstName, lastName) => {
  let response = await fetch(`${baseUrl}/players/?search=${firstName}%20${lastName}&per_page=100`);

  if(!response.ok) {

  }

  let data = await response.json();

  return data.data
}

export const getPlayerSeasonAvgs = async (id) => {
  let response = await fetch(`${baseUrl}/season_averages?player_ids[]=${id}`);

  if(!response.ok) {

  }

  let data = await response.json();

  return data.data;
}

export const getGameByGameStats = async (id) => {
  let response = await fetch(`${baseUrl}/stats?per_page=100&seasons[]=2019&player_ids[]=${id}`)

  if(!response.ok) {

  }

  let data = await response.json();
  console.log(data.data)

  return data.data;
}