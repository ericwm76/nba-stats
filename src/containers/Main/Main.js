import './Main.scss';
import React from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {  } from '../../actions';
import PropTypes from 'prop-types';

export const Main = () => {


  return (
    <div className="Main">
      
    </div>
  )
}

const mapStateToProps = state => ({

})

const mapDispatchToProps = dispatch => (
  bindActionCreators({

  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(Main)