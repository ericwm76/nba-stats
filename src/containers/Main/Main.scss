@import url('https://fonts.googleapis.com/css?family=Rajdhani:400,700&display=swap');

.Main {
  background-color: #212121;
  color: white;
  font-family: "Rajdhani", sans-serif;  
  height: 90vh;
  padding: 20px;
  text-align: center;
  button {
    background-color: #212121;
    border: 1px solid white;
    border-radius: 3px;
    color: white;
    display: block;
    font-family: "Rajdhani", sans-serif;  
    font-size: .9em;
    height: 44px; 
    margin: 5px;
    width: 10%;
  }
}