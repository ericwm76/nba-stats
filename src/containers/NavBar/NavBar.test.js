import React from 'react';
import { shallow } from 'enzyme';
import { Main, mapStateToProps } from './Main';
import { getGameByGameStats } from '../../apiCalls/apiCalls';