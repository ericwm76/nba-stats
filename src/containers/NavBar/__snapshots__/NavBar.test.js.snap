// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<div
  className="NavBar"
>
  <h1>
    NBA Stats
  </h1>
  <form>
    <input
      className="firstNameInput"
      name="firstName"
      onChange={[Function]}
      placeholder="Player's first name, e.g. LeBron"
      type="text"
    />
    <input
      className="lastNameInput"
      name="lastName"
      onChange={[Function]}
      placeholder="Player's last name, e.g. James"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </form>
</div>
`;
